src/controllers/api/missionInventoryUpdateController.ts

```diff
@@ -50,13 +63,133 @@ const missionInventoryUpdateController: RequestHandler = async (req, res) => {
+            // InventoryJson, // this part will reset game data and missions will be locked
+            TotalCredits,
+            CreditsBonus,
+            MissionCredits
+        });
     } catch (err) {
         console.error("Error parsing JSON data:", err);
     }
+};
+
+/*
+**** OUTPUT ****
+- [x]  InventoryJson
+- [ ]  MissionRewards
+- [x]  TotalCredits
+- [x]  CreditsBonus
+- [x]  MissionCredits
+- [ ]  InventoryChanges
+- [ ]  FusionPoints int
+*/
+
+interface StringDictionary {
+    [key: string]: string;
+}
+const getRewards = (rewardInfo: MissionInventoryUpdateRewardInfo | undefined) => {
+    if (!rewardInfo) return;
+
+    const missionName = (missionNames as StringDictionary)[rewardInfo.node];
+    const rewards = missionsDropTable.find(i => i.mission === missionName)?.rewards;
+
+    if (!rewards) return [];
+
+    // TODO - add Rotation logic
+
+    // Separate guaranteed and chance drops
+    const guaranteedDrops: Reward[] = [];
+    const chanceDrops: Reward[] = [];
+    for (const reward of rewards) {
+        if (reward.chance === 100) guaranteedDrops.push(reward);
+        else chanceDrops.push(reward);
+    }
+
+    const randomDrop = getRandomRewardByChance(chanceDrops);
+    if (randomDrop) guaranteedDrops.push(randomDrop);
+
+    return formatRewardsToInventoryType(guaranteedDrops);
+};
+
+interface Reward {
+    name: string;
+    chance: number;
+    rotation?: string;
+}
+const getRandomRewardByChance = (data: Reward[] | undefined): Reward | undefined => {
+    if (!data || data.length == 0) return;
+
+    const totalChance = data.reduce((sum, item) => sum + item.chance, 0);
+    const randomValue = Math.random() * totalChance;
+
+    let cumulativeChance = 0;
+    for (const item of data) {
+        cumulativeChance += item.chance;
+        if (randomValue <= cumulativeChance) {
+            return item;
+        }
+    }
+
+    return;
+};
+
+const formatRewardsToInventoryType = (rewards: Reward[]) => {
+    return rewards.map(i => {
+        const mod = (modNames as StringDictionary)[i.name];
+        const skin = (skinNames as StringDictionary)[i.name];
+        const gear = (gearNames as StringDictionary)[i.name];
+        const arcane = (arcaneNames as StringDictionary)[i.name];
+        const craft = (craftNames as StringDictionary)[i.name];
+        const misc =
+            (miscNames as StringDictionary)[i.name] || (miscNames as StringDictionary)[i.name.replace(/\d+X\s*/, "")];
+        const resource =
+            (resourceNames as StringDictionary)[i.name] ||
+            (resourceNames as StringDictionary)[i.name.replace(/\d+X\s*/, "")];
+        const relic =
+            (relicNames as StringDictionary)[i.name.replace("Relic", "Exceptional")] ||
+            (relicNames as StringDictionary)[i.name.replace("Relic (Radiant)", "Radiant")];
+
+        let ItemType: string = mod;
+        const ItemCount = 1;
 
-    // TODO - get original response
-    res.json({});
+        if (mod) {
```

Comment:
is this the best way? looks a bit suspicious with the many else ifs

Reply:
need that like this for now